Supervision Work/src/uk/ac/cam/ap886/oopjava/supervisionProjects/Hangman/GUIMainWindow.java

```diff
@@ -0,0 +1,203 @@
+package uk.ac.cam.ap886.oopjava.supervisionProjects.Hangman;
+
+import java.awt.ComponentOrientation;
+import java.awt.Dimension;
+import java.awt.EventQueue;
+import java.awt.Font;
+
+import javax.swing.BoxLayout;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JOptionPane;
+import javax.swing.JPanel;
+import javax.swing.JSeparator;
+
+public class GUIMainWindow implements UserInterface{
+	Game game;
+	boolean firstRun=true;
+	
+	JLabel lblName;
+	JLabel lblGamesWon;
+	JLabel lblLettersTried;
+	JLabel openedWord;
+	JLabel lblStatus;
+	GUIGamePanel hangmanPanel;
+
+	public GUIMainWindow (Game newGame){
+		game = newGame;
+		initialize();
+	}
+
+	private JFrame frame;
+	private JSeparator separator;
+
+	/**
+	 * Launch the application.
+	 */
+	public static void main(String[] args) {
+		EventQueue.invokeLater(new Runnable() {
+			@Override
+			public void run() {
+				try {
+					GUIMainWindow window = new GUIMainWindow();
+					window.frame.setVisible(true);
+				} catch (Exception e) {
```

Comment:
Catch all exception handlers... Not too bad since its in main method but I would argue that if the program crashes here, let it crash, what's the point in catching any old exception bubbled to the top if you're going to terminate anyway?